docs/sytest.md

```diff
@@ -1,8 +1,41 @@
 # SyTest
 
 Dendrite uses [SyTest](https://github.com/matrix-org/sytest) for its
-integration testing. When creating a new PR, add the test IDs that your PR
-should allow to pass to `testfile` in dendrite's root directory. Not all PRs
-need to make new tests pass. If we find your PR should be making a test pass we
-may ask you to add to that file, as generally Dendrite's progress can be
+integration testing. When creating a new PR, add the test IDs (see below) that
+your PR should allow to pass to `testfile` in dendrite's root directory. Not all
+PRs need to make new tests pass. If we find your PR should be making a test pass
+we may ask you to add to that file, as generally Dendrite's progress can be
 tracked through the amount of SyTest tests it passes.
+
+## Finding out which tests to add
```

Comment:
This section is definitely needed, but I think we should encourage people to run tests locally instead of using a CI service (especially as we may change the CI service unexpectedly, like Synapse just did [CircleCI -> BuildKite for Sytests]).

People just need to download and setup [Sytest](https://github.com/matrix-org/sytest/)'s develop branch next to their dendrite checkout:

```
git clone -b develop https://github.com/matrix-org/sytest
cd sytest
./install-deps.pl
```

Set up the DB:

```
sudo -u postgres psql -c "CREATE USER dendrite PASSWORD 'itsasecret'"
sudo -u postgres psql -c "CREATE DATABASE sytest_template OWNER dendrite"
mkdir -p "server-0"
cat > "server-0/database.yaml" << EOF
args:
    user: $PGUSER
    database: $PGUSER
    host: $PGHOST
type: pg
EOF
```

Then run the tests:

```
./run-tests.pl -I Dendrite::Monolith -d ../dendrite/bin -W ../dendrite/testfile -O tap --all | tee results.tap
```

(`tee` lets people see the test results while they're being piped to the file)

Reply:
I think it may be a good idea to also add a guide for docker? I actually ran tests in docker on my local machine.

Reply:
Good idea. In that case, we may want to also add running your script to https://github.com/matrix-org/sytest/blob/develop/docker/dendrite_sytest.sh#L55, so that it's run by the docker container automatically.

So something like:

```
/src/show-expected-fail-tests.sh results.tap
```

Reply:
Should be done now. Waiting for https://github.com/matrix-org/sytest/pull/626.